python/lsst/ts/externalscripts/auxtel/latiss_base_align.py

```diff
@@ -880,3 +887,37 @@ async def run(self) -> None:
         await self.assert_feasibility()
 
         await self.arun(True)
+
+    async def cleanup(self):
+        if self.state.state != ScriptState.ENDING:
+            # abnormal termination
+            if self.iterations_started and self.iterations_executed != self.max_iter:
+                self.log.warning(
+                    f"Terminating with state={self.state.state}: stop iterations. "
+                    f"Iterations started: {self.iterations_started}."
+                )
+                try:
+                    self.log.warning(
+                        f"Loop failed at iteration {self.iterations_executed} from a total of "
+                        f"{self.max_iter}. Returning hexapod to its initial position."
+                    )
+
+                    self.rem.ataos.evt_detailedState.flush()
+
+                    # Return hexapod to original position
+                    await self.atcs.offset_aos_lut(
+                        z=-self.offset_total_focus,
```

Comment:
I think this could work if you keep track of the focus offset you apply when taking the intra/extra focal data. 

Basically, before you do [this](https://github.com/lsst-ts/ts_externalscripts/blob/82671cf7a246e1eb74c3a88dbb665dd78ba709cf/python/lsst/ts/externalscripts/auxtel/latiss_base_align.py#L233) and [this](https://github.com/lsst-ts/ts_externalscripts/blob/82671cf7a246e1eb74c3a88dbb665dd78ba709cf/python/lsst/ts/externalscripts/auxtel/latiss_base_align.py#L253) you would do:

```
self.offset_total_focus += self.dz
```

and 

```
self.offset_total_focus += z_offset
```

respectively.